### ISSUE TITLE
compute server: if the external ip address isn't known, try again until it is known

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#8044]
I saw an instance of a compute server who external ip address isn't known, even 45 minutes after startup.  Everything works except DNS, because nothing but DNS relies on needing the external ip address.

The problem is NOT a longterm api issue with google or with the machine.  I tried directly calling the function to get the ip address:
```js
> a = require('../dist/compute/cloud/google-cloud/get-instance')
> await a.default({name:'prod-XXX',zone:'XXX'})
{
  name: 'prod-6168',
  state: 'running',
  internalIp: '10.XXX',
  externalIp: '35XXX',
  creationTimestamp: 2024-11-17T18:49:41.613Z,
  lastStartTimestamp: 2024-11-26T22:54:37.362Z,
  cpuPlatform: 'AMD Rome'
}
> 
```

and, as you can see, it worked fine.

WORKAROUND: the only workaround for now is to restart the compute server.

I just looked further into our code, and the relevant code is in src/packages/server/compute/cloud/google-cloud/index.ts

```ts
  await waitForIp({
    name,
    zone: configuration.zone,
    id: server.id,
    maxTime: 2 * 60 * 1000,
  });
```

Basically, it waits up to 2 minutes to get the ip address, then gives up.   That's probably enough 99% of the time on Google cloud, but maybe not 100% of the time.  Also the waitForIp function would fail if there is a single intermittent google cloud api call failure, so we should address that.
